cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/KafkaRollerTest.java

```diff
@@ -158,7 +158,7 @@ public void pod1NotReadyAfterRolling(VertxTestContext testContext) throws Interr
         clearRestarted();
         doFailingRollingRestart(testContext, kafkaRoller,
                 asList(2, 3, 4),
-                KafkaRoller.FatalProblem.class, "Error while waiting for non-restarted pod c-kafka-1 to become ready",
+                KafkaRoller.FatalProblem.class, "Error while waiting for restarted pod c-kafka-1 to become ready",
```

Comment:
I changed two assertions in this test because the vertx-junit5 upgrade revealed that the tests were false positives. Debugging them on vertx-4.2.4 showed that the assertions inside `testContext.failing` were ignored.

The test expected them to be non-restarted but the roller decides they need reconfiguration due to logging config difference.

Reply:
This test class looks like it has similar logic to the one in Topic Operator where functions inside the test function are passed the test context and create checkpoints and call context.CompleteNow. That's dangerous because it's likely that the test is completed when the first call to doFailingRollingRestart is called and it never gets to the second call. This is the PR where I made changes to the similar topic tests: https://github.com/strimzi/strimzi-kafka-operator/pull/7338